actors/verifreg/tests/harness/mod.rs

```diff
@@ -233,13 +236,16 @@ impl Harness {
 
         let params = AddVerifiedClientParams { address: *client, allowance: allowance.clone() };
 
-        rt.expect_emitted_event(
-            EventBuilder::new()
-                .typ("verifier-balance")
-                .field_indexed("verifier", &verifier.id().unwrap())
-                .field("balance", &BigIntSer(&(verifier_balance - allowance)))
-                .build()?,
-        );
+        if client_resolved.id().is_ok() {
```

Comment:
This should probably just unwrap. If the client can't be resolved, something is wrong with the test code and it should abort (e.g. the datacap transfer should also fail).

Reply:
> `thread 'clients::rejects_unresolved_address' panicked at 'called `Result::unwrap()` on an `Err` value: NonIDAddress', actors/verifreg/tests/harness/mod.rs:242:64`

That's why it's in here. But now I don't understand why it's OK to `if` the emit and not have an event in this case.